src/Microsoft.AspNetCore.Server.IISIntegration/IISHttpContextExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Extensions to <see cref="HttpContext"/> that enable access to IIS features.
+    /// </summary>
+    public static class IISHttpContextExtensions
+    {
+        /// <summary>
+        /// Gets the value of a server variable for the current request.
+        /// </summary>
+        /// <param name="context">The http context for the request.</param>
+        /// <param name="variableName">The name of the variable.</param>
+        /// <returns>
+        /// <c>null</c> if the feature does not support the <see cref="IServerVariablesFeature"/> feature.
+        /// May return null or empty if the variable does not exist or is not set.
+        /// </returns>
+        /// <remarks>
+        /// For a list of common server variables available in IIS, see http://go.microsoft.com/fwlink/?LinkId=52471.
+        /// </remarks>
+        public static string GetIISServerVariable(this HttpContext context, string variableName)
```

Comment:
I don't feel like this should show up on the HttpContext for every single ASP.NET Core 2.1 application. If we're going to keep it, lets change the namespace.

Reply:
> Microsoft.AspNetCore.Server.IIS

Yes, that can be the namespace.